transport-classes-io_uring/src/main/java/io/netty/channel/uring/IoUringMetric.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2025 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel.uring;
+
+import io.netty.util.internal.LongCounter;
+import io.netty.util.internal.PlatformDependent;
+import io.netty.util.internal.SystemPropertyUtil;
+
+import java.util.Map;
+
+/**
+ * Metric for {@link IoUring}.
+ */
+public final class IoUringMetric {
+
+    private static final boolean ENABLE_METRIC = SystemPropertyUtil.getBoolean("io.netty.iouring.enableMetric", true);
+
+    private static final LongCounter SQE_HANDLE_COUNTER = PlatformDependent.newLongCounter();
+
+    private static final LongCounter CQE_HANDLE_COUNT = PlatformDependent.newLongCounter();
+
+    private static final LongCounter PROVIDER_BUFFER_READ_FAIL_COUNT = PlatformDependent.newLongCounter();
+
+    /**
+     * Use map instead of directly using CompletionQueue to prevent illegal access after ioUring is closed
+     */
+    private static final Map<CompletionQueue, Integer> OVERFLOW_RECORD = PlatformDependent.newConcurrentHashMap();
+
+    private static final LongCounter OVERFLOW_FROM_CLOSED_CQE = PlatformDependent.newLongCounter();
```

Comment:
Most of the interactions with the ring as single threaded so you can just use AtomicLong counters which use the pattern: `counter.lazySet(counter.get() + 1)`

it will still be atomic, but without any strong mem barrier on increment

Reply:
Okay, I will switch the io.netty.util.internal.LongCounter here to AtomicLong.